src/main/java/htsjdk/samtools/cram/compression/rans/ransnx16/RANSNx16Decode.java

```diff
@@ -0,0 +1,488 @@
+                for (int r=0; r<= run; r++){
+                    rleOutBuffer.put(j++, sym);
+                }
+            }else {
+                rleOutBuffer.put(j++, sym);
+            }
+        }
+        inBuffer = rleOutBuffer;
+        return inBuffer;
+    }
+
+    private ByteBuffer decodePack(
+            ByteBuffer inBuffer,
+            final int[] packMappingTable,
+            final int numSymbols,
+            final int uncompressedPackOutputLength) {
+        ByteBuffer outBufferPack = ByteBuffer.allocate(uncompressedPackOutputLength);
+        int j = 0;
+
+        if (numSymbols <= 1) {
+            for (int i=0; i < uncompressedPackOutputLength; i++){
+                outBufferPack.put(i, (byte) packMappingTable[0]);
+            }
+        }
+
+        // 1 bit per value
+        else if (numSymbols <= 2) {
+            int v = 0;
+            for (int i=0; i < uncompressedPackOutputLength; i++){
+                if (i % 8 == 0){
+                    v = inBuffer.get(j++);
+                }
+                outBufferPack.put(i, (byte) packMappingTable[v & 1]);
+                v >>=1;
+            }
+        }
+
+        // 2 bits per value
+        else if (numSymbols <= 4){
+            int v = 0;
+            for(int i=0; i < uncompressedPackOutputLength; i++){
+                if (i % 4 == 0){
+                    v = inBuffer.get(j++);
+                }
+                outBufferPack.put(i, (byte) packMappingTable[v & 3]);
+                v >>=2;
+            }
+        }
+
+        // 4 bits per value
+        else if (numSymbols <= 16){
+            int v = 0;
+            for(int i=0; i < uncompressedPackOutputLength; i++){
+                if (i % 2 == 0){
+                    v = inBuffer.get(j++);
+                }
+                outBufferPack.put(i, (byte) packMappingTable[v & 15]);
+                v >>=4;
+            }
+        }
+        inBuffer = outBufferPack;
+        return inBuffer;
+    }
+
+    private ByteBuffer decodeStripe(final ByteBuffer inBuffer, final int outSize){
+        final int numInterleaveStreams = inBuffer.get() & 0xFF;
+
+        // retrieve lengths of compressed interleaved streams
+        final int[] compressedLengths = new int[numInterleaveStreams];
+        for ( int j=0; j<numInterleaveStreams; j++ ){
+            compressedLengths[j] = Utils.readUint7(inBuffer);
+        }
+
+        // Decode the compressed interleaved stream
+        final int[] uncompressedLengths = new int[numInterleaveStreams];
+        final ByteBuffer[] TransposedData = new ByteBuffer[numInterleaveStreams];
+        for ( int j=0; j<numInterleaveStreams; j++){
+            uncompressedLengths[j] = (int) Math.floor(((double) outSize)/numInterleaveStreams);
+            if ((outSize % numInterleaveStreams) > j){
+                uncompressedLengths[j]++;
+            }
+
+            TransposedData[j] = uncompress(inBuffer, uncompressedLengths[j]);
```

Comment:
Here is another nested uncompress call - we need to make sure that we're not clobbering any state as mentioned elsewhere in these review comments.

```suggestion
            transposedData[j] = uncompress(inBuffer, uncompressedLengths[j]);
```